definitions/3.0.0/multiFormatSchema.json

```diff
@@ -0,0 +1,147 @@
+                "application/vnd.aai.asyncapi;version=2.1.0",
+                "application/vnd.aai.asyncapi+json;version=2.1.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.1.0",
+                "application/vnd.aai.asyncapi;version=2.2.0",
+                "application/vnd.aai.asyncapi+json;version=2.2.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.2.0",
+                "application/vnd.aai.asyncapi;version=2.3.0",
+                "application/vnd.aai.asyncapi+json;version=2.3.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.3.0",
+                "application/vnd.aai.asyncapi;version=2.4.0",
+                "application/vnd.aai.asyncapi+json;version=2.4.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.4.0",
+                "application/vnd.aai.asyncapi;version=2.5.0",
+                "application/vnd.aai.asyncapi+json;version=2.5.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.5.0",
+                "application/vnd.aai.asyncapi;version=2.6.0",
+                "application/vnd.aai.asyncapi+json;version=2.6.0",
+                "application/vnd.aai.asyncapi+yaml;version=2.6.0",
+                "application/vnd.aai.asyncapi;version=3.0.0",
+                "application/vnd.aai.asyncapi+json;version=3.0.0",
+                "application/vnd.aai.asyncapi+yaml;version=3.0.0"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://asyncapi.com/definitions/3.0.0/schema.json"
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "required": [
+            "schemaFormat"
+          ],
+          "properties": {
+            "schemaFormat": {
+              "enum": [
+                "application/schema+json;version=draft-07",
+                "application/schema+yaml;version=draft-07"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://json-schema.org/draft-07/schema"
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "required": [
+            "schemaFormat"
+          ],
+          "properties": {
+            "schemaFormat": {
+              "enum": [
+                "application/vnd.oai.openapi;version=3.0.0", 
+                "application/vnd.oai.openapi+json;version=3.0.0", 
+                "application/vnd.oai.openapi+yaml;version=3.0.0"
+              ]
+            }
+          }
+        },
+        "then": {
+          "properties": {
+            "schema": {
+              "$ref": "http://asyncapi.com/definitions/3.0.0/openapiSchema_3_0.json"
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "required": [
+            "schemaFormat"
+          ],
+          "properties": {
```

Comment:
If we need to add official support for protobuf, indeed this should be done inside this document by adding a new "if" section. But afaik  Otherwise, it is still supported since the schema for the object itself allows it since the `schema` has no JSON Schema definition:

```json
"properties": {
      "schemaFormat": {
        "type": "string"
      },
      "schema": {}
    },
```

Meaning the following should be valid: 

```json
"components": {
    "schemas": {
      "myProtobufSchema": {
        "schemaFormat": "application/vnd.google.protobuf;version=3",
        "schema": "message Point {\n    required int32 x = 1;\n    required int32 y = 2;\n    optional string label = 3;\n}\n\nmessage Line {\n    required Point start = 1;\n    required Point end = 2;\n    optional string label = 3;\n}\n"
      }
    }
  }
```

Reply:
the `"schema": {}` is the same as `"schema": true`, so everything is allowed.

Reply:
Testing, I noticed there is a bug somewhere in my current implementation where it is allowed to set a string schema even when setting the schemaFormat to be an AsyncAPI schema. 

The following example demonstrates it; where the `testinline` schema appears as invalid (as I expect) but the `test` schema is valid (unexpected). 

```json
{
  "asyncapi": "3.0.0",
  "info": {
    "title": "Example",
    "version": "0.1.0"
  },
  "components": {
    "schemas": {
      "test": {
        "schemaFormat": "application/vnd.aai.asyncapi+json;version=3.0.0",
        "schema": "message Point {\n    required int32 x = 1;\n    required int32 y = 2;\n    optional string label = 3;\n}\n\nmessage Line {\n    required Point start = 1;\n    required Point end = 2;\n    optional string label = 3;\n}\n"
      },
      "testInline": "message Point {\n    required int32 x = 1;\n    required int32 y = 2;\n    optional string label = 3;\n}\n\nmessage Line {\n    required Point start = 1;\n    required Point end = 2;\n    optional string label = 3;\n}\n"
    }
  }
}
```

[Open in JSON Schema validator](https://www.jsonschemavalidator.net/s/8bpDkYR3)

I don't get why since both are validating against `schema.json`. I will need to check what's the issue about.

Reply:
@smoya remember to change the components to be `anySchema.json` not `schema.json` 😄

Reply:
![oh-no-facepalm-gif](https://github.com/asyncapi/spec-json-schemas/assets/1083296/36bba03f-61e1-484e-a70b-a6081603182e)